diffprivlib/utils.py

```diff
@@ -89,6 +90,56 @@ def warn_unused_args(args):
                       "warning." % arg, DiffprivlibCompatibilityWarning)
 
 
+def check_epsilon_delta(epsilon, delta):
+    """Checks that epsilon and delta are valid values for differential privacy. Throws an error if checks fail,
+    otherwise returns nothing.
+
+    As well as the requirements of epsilon and delta separately, both cannot be simultaneously zero.
+
+    Parameters
+    ----------
+    epsilon : float
+        Epsilon parameter for differential privacy. Must be non-negative.
+
+    delta : float
+        Delta parameter for differential privacy. Must be on the unit interval, [0, 1].
+
+    """
+    if not isinstance(epsilon, Real) or not isinstance(delta, Real):
+        raise TypeError("Epsilon and delta must be numeric")
+
+    if epsilon < 0:
+        raise ValueError("Epsilon must be non-negative")
+
+    if not 0 <= delta <= 1:
+        raise ValueError("Delta must be in [0, 1]")
+
+    if epsilon + delta == 0:
+        raise ValueError("Epsilon and Delta cannot both be zero")
+
+
+def check_accountant(accountant):
```

Comment:
I would move this function to a class method

Reply:
I've removed this checker for now (b6ef353) in favour of documentation-led checking. By defaulting to a `BudgetAccountant` instance, this should be sufficient.

Reply:
As of 2bc02c1, the type-checking has been integrated within the `check` method.